deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvDeploys.java

```diff
@@ -218,7 +218,17 @@ public Response create(
         if(buildBean.getScm_branch().equals("private") && ! envBean.getAllow_private_build()) {
             throw new TeletaanInternalException(Response.Status.BAD_REQUEST, String.format("This stage does not allow deploying a private build. Please Contact #teletraan and #security-related to whitelist your stage for deploying private build"));
         }
+
+        // check if the build is from a valid artifact url
+        if(!buildBean.getArtifact_url().startsWith("https://soxrepo.pinadmin.com") && !buildBean.getArtifact_url().startsWith("https://deployrepo.pinadmin.com/")) {
+            throw new TeletaanInternalException(Response.Status.BAD_REQUEST, String.format("This stage does not allow deploying a build from invalid url. Please use a valid url either from deployrepo or soxrepo!"));
+        }
         
+        // check if the build is from a trusted artifact url
+        if(buildBean.getArtifact_url().startsWith("https://soxrepo.pinadmin.com") && !envBean.getEnsure_trusted_build()) {
```

Comment:
should it be:   if (envBean.getEnsure_trusted_build() && ! buildBean.getArtifact_url().startsWith("https://soxrepo.pinadmin.com")) .....

Reply:
Good catch, updated.